src/datalad_installer.py

```diff
@@ -563,21 +545,27 @@ class DataladInstaller:
         ],
     )
 
-    def __init__(
-        self,
-        env_write_files: Optional[list[str | Path]] = None,
-        sudo_confirm: SudoConfirm = SudoConfirm.ASK,
-    ) -> None:
-        #: A list of files to which to write ``PATH`` modifications and related
-        #: shell commands
-        self.env_write_files: list[Path]
-        if env_write_files is None:
-            self.env_write_files = []
-        else:
-            self.env_write_files = [Path(p) for p in env_write_files]
-        self.sudo_confirm: SudoConfirm = sudo_confirm
-        #: The default installers to fall back on for the "auto" installation
-        #: method
+    #: A list of files to which to write ``PATH`` modifications and related
+    #: shell commands
+    env_write_files: list[Path] = field(default_factory=list)
```

Comment:
how they will become `Path` here if `str` would be provided? (I just don't see any relevant changes in this diff)